clients/src/main/resources/common/message/README.md

```diff
@@ -75,16 +75,24 @@ There are several primitive field types available.
 
 * "int16": a 16-bit integer.
 
+* "uint16": a 16-bit unsigned integer.
+
 * "int32": a 32-bit integer.
 
+* "uint32": a 32-bit unsigned integer.
+
 * "int64": a 64-bit integer.
 
 * "float64": is a double-precision floating point number (IEEE 754).
 
 * "string": a UTF-8 string.
 
+* "uuid": a type 4 immutable universally unique identifier.
+
 * "bytes": binary data.
 
+* "records": memory record set or file record set.
```

Comment:
the supported records type in KRPC is only `MemoryRecords`, right?

Reply:
Only `MemoryRecords` are supported when receiving records, but we can also set `FileRecords` when writing records. foe example `KafkaRaftClient.tryCompleteFetchRequest()` will call `partitionData.setRecordSet(log.read(xxx))`

Reply:
There are many sub classes for this type as the java type used by KRPC is `BaseRecords`. We do make protocol data carry different sub class in processing data (before serialization). Hence, it seems to me listing sub classes in this document is a bit weird.

Reply:
I think we should avoid listing subclasses since we would have to update docs every time we add a new subclass, how about rewording this to "recordset such as memory recordset"?

Reply:
That LGTM :)